### PR TITLE
FALCON-2246 Api to list all the jobs in the system

### PR DESCRIPTION
**Pull request by PracheerAgarwal**

### CODE REVIEW

**Comment by sandeepSamudrala** [comment]
Just add a unit test in FalconUnit. Rest looks fine.

File: cli/src/main/java/org/apache/falcon/cli/FalconExtensionCLI.java [hunk 1]
```diff
@@ -51,6 +51,7 @@
 
     // Extension commands
     private static final String ENUMERATE_OPT = "enumerate";
+    private static final String JOB_LIST_OPT = "jobList";
```
Comment (PracheerAgarwal) [inline cli/src/main/java/org/apache/falcon/cli/FalconExtensionCLI.java]: correctly said so, have revert my changes and have modified the existing list API.

File: prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java [hunk 2]
```diff
@@ -106,15 +108,35 @@ public ExtensionJobList getExtensionJobs(
...
+            if (extensionName != null) {
+                extensionJobs = ExtensionStore.getMetaStore().getJobsForAnExtension(extensionName);
+            } else {
+                extensionJobs = ExtensionStore.getMetaStore().getAllExtensionJobs();
+            }
+
             switch (sortOrder.toLowerCase()) {
             case DESCENDING_SORT_ORDER:
-                Collections.sort(jobNames, Collections.reverseOrder(String.CASE_INSENSITIVE_ORDER));
+                Collections.sort(extensionJobs, new Comparator<ExtensionJobsBean>() {
```
Comment (pallavi-rao) [inline prism/src/main/java/org/apache/falcon/resource/proxy/ExtensionManagerProxy.java]: Instead, can we have ExtensionJobsBean implement Comparable interface?

File: prism/src/main/java/org/apache/falcon/resource/AbstractExtensionManager.java [hunk 3]
```diff
@@ -100,6 +107,39 @@ public APIResult getExtensions() {
...
+            @Override
+            public int compare(ExtensionJobsBean o1, ExtensionJobsBean o2) {
+                return o1.getJobName().compareToIgnoreCase(o2.getJobName());
+            }
+        };
+
+        Map<String, String> jobAndExtensionNames = new HashMap<>();
+        List<ExtensionJobsBean> extensionJobs = null;
+        if (extensionName != null) {
+            extensionJobs = new ArrayList<>(ExtensionStore.getMetaStore().getJobsForAnExtension(extensionName));
```
Comment (sandeepSamudrala) [inline prism/src/main/java/org/apache/falcon/resource/AbstractExtensionManager.java]: what if the result set is 0 or null?

**Pull request closed without merging**